Automattic/newspack-blocks
# Donate block – mobile style; UX tweaks

### All Submissions:






### Changes proposed in this Pull Request:



Some UX and mobile-styling tweaks to the Donate block, and a conversion to TypeScript. 

Closes #1077

### How to test the changes in this Pull Request:

1. Insert a Donate block, observe the donation frequency tabs are in one line on all screen sizes (not stacked, as they used to be) 
2. Observe that the "Other" donation tier has a minimum of 1 (e.g. when using arrow keys, it will stop at 1)
3. With Stripe set as Reader Revenue platform:
    1. Observe that if card payment is the only available option, the card UI will be visible right away
    2. If payment request (Apple/Google/Web pay) is available, the card UI will be displayed only after clicking "Donate with card" button

### Other information:
